runtime/src/main/cpp/Runtime.cpp

```diff
@@ -149,6 +186,63 @@ void Kotlin_deinitRuntimeIfNeeded() {
   }
 }
 
+// TODO: Consider exporting it to interop API.
+void Kotlin_shutdownRuntime() {
+    // TODO: If checkers are disabled, we can set status to "shutdown" here, and return.
+    auto lastStatus = compareAndSwap(&globalRuntimeStatus, kGlobalRuntimeRunning, kGlobalRuntimeShuttingDown);
+    switch (lastStatus) {
+        case kGlobalRuntimeRunning:
+            break;
+        case kGlobalRuntimeShuttingDown:
+        case kGlobalRuntimeShutdown:
+            konan::consoleErrorf("Cannot shutdown Kotlin runtime twice\n");
+            konan::abort();
+        case kGlobalRuntimeUninitialized:
+            konan::consoleErrorf("Kotlin runtime must have been initialized\n");
+            konan::abort();
+    }
+
+    auto* runtime = ::runtimeState;
+    if (runtime == kInvalidRuntime) {
+        konan::consoleErrorf("Current thread must have Kotlin runtime initialized on it\n");
+        konan::abort();
+    }
+
+    if (Kotlin_cleanersLeakCheckerEnabled()) {
+        // Make sure to collect any lingering cleaners.
+        PerformFullGC(runtime->memoryState);
+    }
+
+    // Stop cleaner worker. Only execute the cleaners if checker is enabled.
+    ShutdownCleaners(Kotlin_cleanersLeakCheckerEnabled());
+
+    // Cleaners are now done, disallow new runtimes.
+    lastStatus = compareAndSwap(&globalRuntimeStatus, kGlobalRuntimeShuttingDown, kGlobalRuntimeShutdown);
+    RuntimeAssert(lastStatus == kGlobalRuntimeShuttingDown, "Must be in ShuttingDown state");
+
+    // Spin until all runtimes have fully initialized.
+    while (ScopedInitializingRuntime::IsInitializing()) {
+    }
+
+    // TODO: If we add early return at the top, this if would be unneeded.
+    if (Kotlin_memoryLeakCheckerEnabled() || Kotlin_memoryLeakCheckerEnabled()) {
```

Comment:
Why check the same twice?

Reply:
My mistake, meant to check for cleaners checker.